lua/lualine/highlight.lua

```diff
@@ -10,7 +10,41 @@ local function highlight (name, foreground, background, special)
   return table.concat(command, ' ')
 end
 
+local function apply_defaults_to_theme(theme)
+  local modes = {'insert', 'visual', 'replace', 'command', 'terminal', 'inactive'}
+  -- normal mode cann't have a fallback for now
+  local sections = {'a', 'b', 'c', 'x', 'y', 'z'}
+  for _, mode in ipairs(modes) do
+    repeat
+      if theme[mode] == nil then
+        theme[mode] = theme['normal']
+        break
+      end
+      for _, section in ipairs(sections) do
+        repeat
```

Comment:
I don't understand what is `repeat` and `until true` supposed to do in this situation, could you elaborate? `luackeck` marks `until true` as unreachable code...

Reply:
(repeat ... break.. utill true) it's basically a continue statement(lua doesn't have continue). I didn't like the structure either . Second repeat loop was unnessary .